crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -248,36 +248,47 @@ impl<'a> MoveTestAdapter<'a> for SuiTestAdapter<'a> {
-                SuiInitArgs {
-                    accounts,
-                    protocol_version,
-                    max_gas,
-                },
-            )) => {
-                let map = verify_and_create_named_address_mapping(named_addresses).unwrap();
-                let accounts = accounts
-                    .map(|v| v.into_iter().collect::<BTreeSet<_>>())
-                    .unwrap_or_default();
-                let mut protocol_config = if let Some(protocol_version) = protocol_version {
-                    ProtocolConfig::get_for_version(protocol_version.into(), Chain::Unknown)
-                } else {
-                    ProtocolConfig::get_for_max_version_UNSAFE()
-                };
-                if let Some(mx_tx_gas_override) = max_gas {
-                    protocol_config.set_max_tx_gas_for_testing(mx_tx_gas_override)
+        let (additional_mapping, account_names, protocol_config, version) =
+            match task_opt.map(|t| t.command) {
+                Some((
+                    InitCommand { named_addresses },
+                    SuiInitArgs {
+                        accounts,
+                        protocol_version,
+                        max_gas,
+                    },
+                )) => {
+                    let map = verify_and_create_named_address_mapping(named_addresses).unwrap();
+                    let accounts = accounts
+                        .map(|v| v.into_iter().collect::<BTreeSet<_>>())
+                        .unwrap_or_default();
+                    let version;
+                    let mut protocol_config = if let Some(protocol_version) = protocol_version {
+                        version = protocol_version.into();
+                        ProtocolConfig::get_for_version(protocol_version.into(), Chain::Unknown)
+                    } else {
+                        version = ProtocolVersion::MAX;
+                        ProtocolConfig::get_for_max_version_UNSAFE()
+                    };
+                    if let Some(mx_tx_gas_override) = max_gas {
+                        protocol_config.set_max_tx_gas_for_testing(mx_tx_gas_override)
+                    }
+                    (map, accounts, protocol_config, version)
                 }
-                (map, accounts, protocol_config)
-            }
-            None => (
-                BTreeMap::new(),
-                BTreeSet::new(),
-                ProtocolConfig::get_for_max_version_UNSAFE(),
-            ),
-        };
+                None => {
+                    let protocol_config = ProtocolConfig::get_for_max_version_UNSAFE();
+                    (
+                        BTreeMap::new(),
+                        BTreeSet::new(),
+                        protocol_config,
+                        ProtocolVersion::MAX,
+                    )
+                }
+            };
+
+        if version != 20.into() {
+            protocol_config.shared_object_deletion();
+        }
```

Comment:
Sorry if I missed the comment, but this should be reverted right? Like we should be relying on the protocol config to set this + setting the protocol config via the `init` flag

Reply:
I forgot to respond, it was specifically recommended by @dariorussi that for any transaction tests that were updated to match functionality for protocol v 21, that we add the v 20 test with the explicit version specified in the test file, so that before v21 is enabled, we still have test coverage by all existing tests

Reply:
@laura-makdah we just talked and can you write up the issue here? thanks

Reply:
In the transaction test framework, I have explicitly enabled the shared object deletion flag on the protocol config that is used for those tests. Additionally, the test protocol config that is used in those tests is set up such that it doesn't contain an actual version unless the flag was set in the file like `--protocol-version n`. In order to ensure the correct test behavior for the new functions going forward I set the SharedObjectDeletion flag to true. But until the protocol version is bumped to 21, that would leave us with a testing gap for the changed tests in the current version 20. I mitigated this by adding the exiting test with existing output and specifying version 20 in the file. Since the test protocol config doesn't include a version unless specified, I need to check for version 20 specifically in the test framework here in order to not enable shared object deletion for those version 20 tests. I will remove the version 20 tests after we upgrade to the next version.